### PR TITLE
Migrate cdk to dev dependency

### PR DESCRIPTION
**Pull request by akinsola-guardian**
## What does this change?
- Moved @guardian/cdk to devDependencies
- Updated @guardian/cdk and other peer dependencies (aws-cdk & aws-cdk-lib)
- Added wrap-ansi as resolution
- Defined a package manager and version

Tested in CODE.
## Why?
- Dev-X recommended moving @guardian/cdk to devDependencies
- wrap-ansi was set to 7.0.0 to get around the following issue 
```
Error [ERR_REQUIRE_ESM]: require() of ES Module /home/runner/work/consent-management-platform/consent-management-platform/cdk/node_modules/wrap-ansi/index.js from /home/runner/work/consent-management-platform/consent-management-platform/cdk/node_modules/cliui/build/index.cjs not supported.
Instead change the require of index.js in /home/runner/work/consent-management-platform/consent-management-platform/cdk/node_modules/cliui/build/index.cjs to a dynamic import() which is available in all CommonJS modules.
    at Object.<anonymous> (/home/runner/work/consent-management-platform/consent-management-platform/cdk/node_modules/cliui/build/index.cjs:293:14)
    at Object.<anonymous> (/home/runner/work/consent-management-platform/consent-management-platform/cdk/node_modules/yargs/build/index.cjs:1:60678)
    at Object.<anonymous> (/home/runner/work/consent-management-platform/consent-management-platform/cdk/node_modules/yargs/index.cjs:5:30)
    at _yargs (/home/runner/work/consent-management-platform/consent-management-platform/cdk/node_modules/jest-cli/build/run.js:30:39)
    at buildArgv (/home/runner/work/consent-management-platform/consent-management-platform/cdk/node_modules/jest-cli/build/run.js:149:26)
    at Object.run (/home/runner/work/consent-management-platform/consent-management-platform/cdk/node_modules/jest-cli/build/run.js:124:24)
    at Object.<anonymous> (/home/runner/work/consent-management-platform/consent-management-platform/cdk/node_modules/jest-cli/bin/jest.js:16:17)
    at Object.<anonymous> (/home/runner/work/consent-management-platform/consent-management-platform/cdk/node_modules/jest/bin/jest.js:12:3)
error Command failed with exit code 1.
```

### CODE REVIEW

File: cdk/package.json [hunk 1]
```diff
@@ -33,9 +32,12 @@
...
-		"@guardian/cdk": "53.1.0",
 		"npm": "^10.4.0",
 		"source-map-support": "^0.5.20"
 	},
-	"prettier": "@guardian/prettier"
+	"resolutions": {
+		"wrap-ansi": "7.0.0"
+	},
+	"prettier": "@guardian/prettier",
+	"packageManager": "yarn@1.22.5"
```
Comment (coldlink) [inline cdk/package.json:42]: Is there a specific reason the `packageManager` is needed here?

Comment (coldlink) [reply cdk/package.json:42]: As if a user has a different/newer version it could cause issues. Maybe define it as a range?

Comment (akinsola-guardian) [reply cdk/package.json]: Just removed this.

Comment (akinsola-guardian) [reply cdk/package.json]: thanks

**Review by coldlink — APPROVED** [review]

**Pull request merged**